src/fileloader.h

```diff
@@ -131,6 +134,12 @@ class FileLoader : public QObject
             {
                 g_logger.error("Error loading %s: %s", loadInfo.filePath, e.what());
             }
+
+            if (geom.get())
+                geom->disconnect();
```

Comment:
Seemed to me that geometry loading could (in principle) fail so severely it would make sense to nuke the object.  I don't mind being defensive, or not defensive.